File: src/main/java/org/taktik/icure/logic/AgendaLogic.java
package org.taktik.icure.logic;

import org.taktik.icure.entities.Agenda;
import org.taktik.icure.entities.CalendarItemType;
import org.taktik.icure.exceptions.DeletionException;

import java.util.List;

public interface AgendaLogic extends EntityPersister<Agenda, String>{
    Agenda createAgenda(Agenda agenda);

    List<String> deleteAgenda(List<String> ids) throws DeletionException;

    Agenda getAgenda(String agenda);

    Agenda modifyAgenda(Agenda agenda);

    List<Agenda> getAllAgendaForUser(String userId);

    List<Agenda> getReadableAgendaForUser(String userId);
}

File: src/main/kotlin/org/taktik/icure/asynclogic/AgendaLogic.kt
/*
 *  iCure Data Stack. Copyright (c) 2020 Taktik SA
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU Affero General Public License as
 *     published by the Free Software Foundation, either version 3 of the
 *     License, or (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful, but
 *     WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *     Affero General Public License for more details.
 *
 *     You should have received a copy of the GNU Affero General Public
 *     License along with this program.  If not, see
 *     <https://www.gnu.org/licenses/>.
 */

package org.taktik.icure.asynclogic

import kotlinx.coroutines.flow.Flow
import org.taktik.couchdb.DocIdentifier
import org.taktik.icure.entities.Agenda

interface AgendaLogic : EntityPersister<Agenda, String> {
    suspend fun createAgenda(agenda: Agenda): Agenda?
    fun deleteAgenda(ids: List<String>): Flow<DocIdentifier>

    suspend fun getAgenda(agenda: String): Agenda?
    suspend fun modifyAgenda(agenda: Agenda): Agenda?
    fun getAllAgendaForUser(userId: String): Flow<Agenda>
    fun getReadableAgendaForUser(userId: String): Flow<Agenda>
}
